pkg/controller/node/node_controller.go

```diff
@@ -496,6 +506,88 @@ func (ctrl *Controller) getNodesForPool(pool *mcfgv1.MachineConfigPool) ([]*core
 	return nodes, nil
 }
 
+func updateLabels(node *corev1.Node, mcls []mcfgv1.MachineConfigLabels) *corev1.Node {
+	for _, mcl := range mcls {
+		for label, value := range mcl.Labels {
+			if mcl.Exist {
+				node.ObjectMeta.Labels[label] = value
+			} else {
+				delete(node.ObjectMeta.Labels, label)
+			}
+		}
+	}
+	return node
+}
+
+func findTaint(node *corev1.Node, taint corev1.Taint) (bool, int) {
+	for i, t := range node.Spec.Taints {
+		if t.Key == taint.Key {
+			return true, i
+		}
+	}
+	return false, 0
+}
+
+func updateTaints(node *corev1.Node, mcts []mcfgv1.MachineConfigTaint) *corev1.Node {
+	for i := range mcts {
+		mct := &mcts[i]
+		found, index := findTaint(node, mct.Taint)
+		if mct.Exist {
+			if !found {
+				// Supposed to be there but wasn't found, add it.
+				node.Spec.Taints = append(node.Spec.Taints, mct.Taint)
+				continue
+			}
+			// Supposed to be there and was, just overwrite it, instead of wasting time
+			// checking if it has the right values.
+			node.Spec.Taints[index] = mct.Taint
+		} else {
+			if !found {
+				// Not supposed to be there and not there
+				continue
+			}
+			// Not supposed to be there, but is, so delete it.
+			node.Spec.Taints = append(node.Spec.Taints[:index], node.Spec.Taints[index+1:]...)
+		}
+	}
+	return node
+}
+
+func (ctrl *Controller) syncLabelsAndTaints(oldNode *corev1.Node, currentConfigName string) error {
+	mc, err := ctrl.mcLister.Get(currentConfigName)
+	if err != nil {
+		return err
+	}
+
+	return clientretry.RetryOnConflict(nodeUpdateBackoff, func() error {
+		oldData, err := json.Marshal(oldNode)
+		if err != nil {
+			return err
+		}
+
+		newNode := oldNode.DeepCopy()
```

Comment:
you need a fresh copy of the node object here at every iteration if there's a conflict?

Reply:
Happened to notice the mcd has `updateNodeRetry()`.